Caltech-IPAC/firefly
# FIREFLY-1401: UI Conversion: Improve chart options panel

Fixes [FIREFLY-1401](https://jira.ipac.caltech.edu/browse/FIREFLY-1401) (mostly)

- Removed `type=switch` from CheckboxGroupInputField because it was buggy and switch is meant to be used as singleton (not in group)
- Created SwitchInputField to have a field group connected Joy-UI switch with tooltip and labels and used it in Error and other places using CheckboxGroupInputField type=switch
- Fixed wrong usage of switch in TAP panel and Edit ADQL
- Fixed bug in heatmap options colorscale field
- Used CollapsiblePanelGroup to group Chart and Trace options's collapsibles in different chart modes
- Converted ImageTableRowViewer to joyUI
- Fixed rubin chart#3  bug

## Testing
Firefly: https://fireflydev.ipac.caltech.edu/firefly-1401-improve-chat-opts/firefly/

SphereX: https://firefly-1401-improve-chat-opts.irsakudev.ipac.caltech.edu/applications/spherex/

- Check switch(es) in:
    - Add Chart -> X and Y Error fields
    - TAP Panel -> CADC -> "Enter my URL" switch with helper text below the input box, "Use Image search" switch 
    - Edit ADQL -> insert fully qualified column name with helper text below label
- Go to [firefly chart tests](https://fireflydev.ipac.caltech.edu/firefly-1401-improve-chat-opts/firefly/test/tests-chart.html) -> #7heatmap -> chart options -> colorscale field should show "Greens" (instead of no option selected "Default" - which was the bug)
- Open Add Chart or Chart Options -> Change plot type to Scatter, Heatmap, Histogram -> in all 3 of them, the "Trace Options" and "Chart Options" accordions should be grouped
- In spherex -> make a search (leads to forced photometry) -> select "spectra preview" tab -> the Image Slider at the bottom should look converted and work as before. Additional: also test expanded mode - close button should look better than before.
- Rubin bug as explained in the link in ticket should no longer be there - open chart options, select a column from the suggestion list, the tooltip should disappear, also hover over tooltip will now won't keep it in focus which was making it interfere with mouse movement over other fields
